stakingModel/contracts/UBIScheme.sol

```diff
@@ -247,29 +228,28 @@ contract UBIScheme is AbstractUBI {
         return _claim(msg.sender);
     }
 
-    /* @dev In Order to update users from active to inactive, we give out incentive to people
-     * to update the status of inactive users, this action is called “Fishing”. Anyone can
-     * send a tx to the contract to mark inactive users. The “fisherman” receives a reward
+    /* @dev In order to update users from active to inactive, we give out incentive to people
+     * to update the status of inactive users, this action is called "Fishing". Anyone can
+     * send a tx to the contract to mark inactive users. The "fisherman" receives a reward
      * equal to the daily UBI (ie instead of the “fished” user). User that “last claimed” > 14
-     * can be “fished” and made inactive (reduces active users count by one). Requires
+     * can be "fished" and made inactive (reduces active users count by one). Requires
      * contract to be active.
      * @param account to fish
      * @return A bool indicating if UBI was fished
      */
     function fish(address account)
-        public
+        public // TODO NOT TOO MANY TESTS, NOT CHECKING SO MANY THINGS IN EACH TEST, LOOK AT ETORO X
         requireActive
         returns (bool)
     {
         setDay();
-        require(identity.isWhitelisted(account), "is not whitelisted");
         require(!isNewUser(account), "a new user");
-        require(checkActiveEntitlement(account), "an active user");
-        require(activeUsersAddresses[account], "has already fished");
+        require(!isPendingUser(account), "a pending user");
+        require((!identity.isWhitelisted(account) && activeUsersAddresses[account]) ||
+                (identity.isWhitelisted(account) && !isActiveUser(account)), "is active or fished user");
```

Comment:
doesnt make sense.
I understand we need a way not to double fish a user.
so i think we actually need a "inactiveUser[address]" list and not an activeUsers list since active is defined by lastclaim
I dont see what does whitelisting has to do with any of these. a user can be fished regardless of his whitelisting status

Reply:
if the user was in the whitelist and had been activated but later on he has been removed, then the user still appears the active users list for 14 days (max). during those days other users may receive higher ubi if that user won't be marked as an active user.
on the other hand, if that user is whitelisted we want to give him the grace of 14 days before he becomes an inactive user.

anyhow, I added a blacklist mechanism (inactive list) and changed the logic.